mlflow/entities/run_tag.py

```diff
@@ -1,14 +1,51 @@
+import logging
+
 from mlflow.entities._mlflow_object import _MLflowObject
+from mlflow.environment_variables import MLFLOW_TRUNCATE_LONG_ENTITIES
+from mlflow.exceptions import MlflowException
 from mlflow.protos.service_pb2 import RunTag as ProtoRunTag
+from mlflow.utils.validation import (
+    MAX_ENTITY_KEY_LENGTH,
+    MAX_TAG_VAL_LENGTH,
+    _validate_length_limit,
+    _validate_tag_name,
+)
+
+_logger = logging.getLogger(__name__)
 
 
 class RunTag(_MLflowObject):
     """Tag object associated with a run."""
 
     def __init__(self, key, value):
+        RunTag.validate_key(key)
+        value = RunTag.validate_value(value)
         self._key = key
         self._value = value
 
+    @staticmethod
+    def validate_key(key):
+        _validate_tag_name(key)
+        _validate_length_limit("Tag key", MAX_ENTITY_KEY_LENGTH, key)
+
+    @staticmethod
+    def validate_value(value):
+        length = len(value)
+        if length <= MAX_TAG_VAL_LENGTH:
+            return value
+
+        if MLFLOW_TRUNCATE_LONG_ENTITIES.get():
+            _logger.warning(
+                f"Tag value {value[:250]!r} ({length} characters) is truncated to "
+                f"{MAX_TAG_VAL_LENGTH} characters to meet the length limit."
+            )
```

Comment:
```suggestion
            _logger.warning(
                f"Tag value {value[:250]!r} ({length} characters) is truncated to "
                f"{MAX_TAG_VAL_LENGTH} characters to meet the length limit. To turn this off, blah blah blah"
            )
```

We could include how to turn off auto-truncation in the message, but that would be annoying if you have multiple long values.